task6/script.js

```diff
@@ -0,0 +1,293 @@
+            author: 'Иванов Иван'
+        },
+
+        {
+            id: '16',
+            description: 'Более 76 тыс. человек во всем мире уже излечились от заболевания, спровоцированного новым коронавирусом, тогда как количество смертей превысило 6,4 тыс.',
+            createdAt: new Date('2019-03-17T23:00:00'),
+            author: 'Иванов Иван'
+        },
+
+        {
+            id: '17',
+            description: 'Более 76 тыс. человек во всем мире уже излечились от заболевания, спровоцированного новым коронавирусом, тогда как количество смертей превысило 6,4 тыс.',
+            createdAt: new Date('2019-03-17T23:00:00'),
+            author: 'Иванов Иван'
+        },
+
+        {
+            id: '18',
+            description: 'Более 76 тыс. человек во всем мире уже излечились от заболевания, спровоцированного новым коронавирусом, тогда как количество смертей превысило 6,4 тыс.',
+            createdAt: new Date('2019-03-17T23:00:00'),
+            author: 'Иванов Иван'
+        },
+
+        {
+            id: '19',
+            description: 'Более 76 тыс. человек во всем мире уже излечились от заболевания, спровоцированного новым коронавирусом, тогда как количество смертей превысило 6,4 тыс.',
+            createdAt: new Date('2019-03-17T23:00:00'),
+            author: 'Николай'
+        },
+
+        {
+            id: '20',
+            description: 'Более 76 тыс. человек во всем мире уже излечились от заболевания, спровоцированного новым коронавирусом, тогда как количество смертей превысило 6,4 тыс.',
+            createdAt: new Date('2021-03-17T23:00:00'),
+            author: 'Николай'
+        },
+
+    ];
+
+    
+    function getPost(id) {
+        for (let i = 0; i < posts.length ; i++) {
+            if (posts[i].id === id) {
+                return posts[i];
+            }
+        }
+    }
+    
+    function validatePost(post){
+        if(typeof post.id !== 'string' || post.id === ''){
+            return false;
+        }
+        if(typeof post.description !== 'string' || post.description === '' || post.description.length > 200)
+            return false;
+        if(!(post.createdAt instanceof Date)){
+            return false;
+        }
+        if(typeof post.author !== 'string' || post.author === ''){
+            return false;
+        }
+        return true;
+    }
+    
+    function addPost(post) {
+        if (validatePost(post)) {            
+            posts.push(post);
+            return true;
+        }
+        return false;
+    }
+    
+    function removePost(id){
+        for(let i = 0; i < posts.length; i++) {
+            if(posts[i].id === id){
+                posts.splice(i,1);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    
+    function editPost(id, post){
+        let editPost = getPost(id);
+        if(post.description !== '' && typeof post.description == 'string'){
+            editPost.description = post.description;
```

Comment:
Object is a reference type, so changing it here, will change it in the array. It can lead to case when there is an invalid post in your collection.